### PR TITLE
feat: Add logprobs support to TRTLLM backend

### PR DESCRIPTION
**Pull request by esoba**
#### Overview:
Add logprobs extraction and propagation for TRTLLM backend

#### Details:

- Added `_extract_logprobs()` method to extract and format logprobs from TRTLLM engine output
- Propagates `log_probs` and `top_logprobs` to Dynamo's response format
- Handles `top_logprobs=0` edge case using `max(1, logprobs)`
- Added `ChatPayloadWithLogprobs` and `CompletionPayloadWithLogprobs` test utilities
- Added e2e tests for aggregated and disaggregated logprobs configurations

#### Where should the reviewer start?

- `components/src/dynamo/trtllm/request_handlers/handler_base.py` - main logprobs extraction logic
- `tests/serve/test_trtllm.py` - new test configurations
-  'tests/utils/payloads.py' - Added dataclass for payload w/logits
-  'tests/utils/payload_builder.py' - Added arguments for logprobs to chat/completion payloads

#### Related Issues: (use one of the action keywords Closes / Fixes / Resolves / Relates to)

- closes GitHub issue: #4684 
- Follows approach from #4678 + #4697 

## Summary by CodeRabbit

* **New Features**
  * Added log probability (logprobs) support to retrieve token-level probability information in responses
  * Users can now request logprobs and top_logprobs parameters to obtain probability data for generated tokens and alternative top choices

* **Tests**
  * Added test configurations and utilities for validating logprobs functionality across different parameter combinations

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Comment by rmccorm4** [comment]
/ok to test [111d0e8](https://github.com/ai-dynamo/dynamo/pull/4759/commits/111d0e81330706aff0b9ee340df4485a2c76a57b)

**Comment by rmccorm4** [comment]
@esoba please fix the failing precommit check: https://github.com/ai-dynamo/dynamo/actions/runs/19946850289/job/57198198828?pr=4759 - just some linting

**Comment by tanmayv25** [comment]
The code changes look good to me. But there are still some code formatting errors showing up pre-commit runs.

**Comment by tanmayv25** [comment]
Can you fix the conflict in `payload_builder.py` and failing DCO test?
https://github.com/ai-dynamo/dynamo/pull/4759/checks?check_run_id=57495128760

File: tests/serve/test_trtllm.py [hunk 1]
```diff
@@ -91,6 +94,34 @@ class TRTLLMConfig(EngineConfig):
...
+            chat_payload(content=TEXT_PROMPT, logprobs=False, top_logprobs=5),
+            chat_payload(content=TEXT_PROMPT, logprobs=True, top_logprobs=None),
+            chat_payload(content=TEXT_PROMPT, logprobs=True, top_logprobs=0),
+        ],
+    ),
+    "disaggregated_logprobs": TRTLLMConfig(
+        name="disaggregated_logprobs",
+        directory=trtllm_dir,
+        script_name="disagg.sh",
+        marks=[pytest.mark.gpu_1, pytest.mark.pre_merge, pytest.mark.trtllm],
```
Comment (tanmayv25) [inline tests/serve/test_trtllm.py:115]:
```suggestion
        marks=[pytest.mark.gpu_2, pytest.mark.pre_merge, pytest.mark.trtllm],
```

**Review by tanmayv25 — CHANGES_REQUESTED** [review]
Need to fix the test case

File: tests/serve/test_trtllm.py (hunk 1 above)
Comment (tanmayv25) [reply tests/serve/test_trtllm.py:115]: I see that this is resolve in latest commit.

**Review by tanmayv25 — APPROVED** [review]
